graphql-server/Sources/App/GraphQLHandler.swift

```diff
@@ -0,0 +1,23 @@
+import NIO
+import Graphiti
+import GraphQL
+
+public struct GraphQLHandler {
+    let context = Context()
+    let api: GraphQLAPI
+    let group: EventLoopGroup
+    
+    init(group: EventLoopGroup) throws {
+        self.group = group
+        let resolver = Resolver(group: group)
+        self.api = try GraphQLAPI(resolver: resolver)
+    }
+    
+    /// Executes queries
+    /// - Parameter query: a String with a valid GraphQL query. Like `{ message { context }}`
+    /// - Returns: `EventLoopFuture<GraphQLResult>` which might contain results or be a failure
+    /// Note that Graphiti fails internally on invalid queries like `{ FAIL` returning a `500 Internal Server Error`
+    func handle(query: String) -> EventLoopFuture<GraphQLResult> {
+        api.execute(request: query, context: context, on: group)
+    }
```

Comment:
I'm not sure this is your issue as you are copying the sample from Graphiti. But it is frustrating you cannot control which `EventLoop` the `handle(query)` call occurs on. It is generally best to try and run any database access etc on the same `EventLoop` as the request.

It seems you pass in an `EventLoopGroup` to the `handle` function but then the `Resolver` needs its own reference to the `EventLoopGroup` as well. Not sure why the `EventLoopGroup` passed to the `handle` function can't be passed down to `Resolver.message`. These aren't your issues though this is an issue with the `Graphiti` package.

Reply:
This is suboptimal, but I’d rather land the example and refine this later after chatting with the Graphiti maintainers.